compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -70,6 +70,10 @@ use time::OffsetDateTime;
 use time::macros::format_description;
 use tracing::trace;
 
+#[cfg(feature = "mimalloc")]
+#[global_allocator]
+static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
```

Comment:
Is this just temporary, or the long-term state? As-is this doesn't affect LLVM's malloc calls?

IOW, why is this not changign the rustc/src/main.rs malloc overrides?

Reply:
Long term. It's better than the linker hacks in main.rs.

Overriding the C allocator only in main.rs doesn't affect DLLs making it unsound without further work dealing with loadable backends and LLVM plugins.

Reply:
But isn't this still a regression for LLVM's allocations? It seems odd to measure perf for a halfway shift like this to mimalloc...

Reply:
We'd still want the linker hacks on Linux / macOS for LLVM, using `global_allocator` just makes Rust allocation overriding less hacky.

As for perf on this PR, LLVM uses Windows' heap allocator both before and after.